File: PROGRAMMERS/LEVEL1/예산.cpp
#include <iostream>
#include <stdio.h>
#include <string>
#include <algorithm>
#include <vector>

using namespace std;

int solution(vector<int> d, int budget) {
    int answer = 0;
    int sum = 0;
    sort(d.begin(), d.end());
    if (d[answer] > budget) return 0;
    while (1) {
        sum += d[answer];
        if (answer == d.size() - 1 || (sum + d[answer + 1]) > budget) break;
        answer++;
    }
    answer++;
    return answer;
}

File: PROGRAMMERS/LEVEL1/예산.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.*;

class Solution {
    public int solution(int[] d, int budget) {
        int answer = 0;
        int sum=0;
        
        Arrays.sort(d);
        if(d[answer]>budget) return 0;
        while(true) {
        	sum+=d[answer];
        	if(answer==d.length-1 || (sum+d[answer+1])>budget) {
        		break;
        	}
        	answer++;
        }
        answer++;
        
        return answer;
    }
}